### PR TITLE
Enable E2E deploy tests on publish

### PR DESCRIPTION
**Pull request by ijjk**
This enables testing against Vercel deployments whenever we publish a new canary to ensure we aren't regressing there. These tests can also be run locally with a Vercel token with `VERCEL_TEST_TOKEN=<token> NEXT_TEST_MODE=deploy node run-tests.js --type e2e`

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js tests/enable-deploy-e2e | Change |
| - | - | - | - |
| buildDuration | 19.6s | 20.3s | ⚠️ +682ms |
| buildDurationCached | 7.7s | 7.8s | ⚠️ +158ms |
| nodeModulesSize | 478 MB | 478 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js tests/enable-deploy-e2e | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 5.412 | 5.157 | -0.25 |
| / avg req/sec | 461.98 | 484.76 | +22.78 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.132 | 2.118 | -0.01 |
| /error-in-render avg req/sec | 1172.48 | 1180.13 | +7.65 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js tests/enable-deploy-e2e | Change |
| - | - | - | - |
| 925.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42 kB | 42 kB | ✓ |
| main-HASH.js gzip | 28.8 kB | 28.8 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 72.4 kB | 72.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js tests/enable-deploy-e2e | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js tests/enable-deploy-e2e | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.36 kB | 1.36 kB | ✓ |
| _error-HASH.js gzip | 193 B | 193 B | ✓ |
| amp-HASH.js gzip | 308 B | 308 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 2.71 kB | 2.71 kB | ✓ |
| head-HASH.js gzip | 359 B | 359 B | ✓ |
| hooks-HASH.js gzip | 920 B | 920 B | ✓ |
| image-HASH.js gzip | 5.72 kB | 5.72 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.65 kB | 2.65 kB | ✓ |
| routerDirect..HASH.js gzip | 320 B | 320 B | ✓ |
| script-HASH.js gzip | 391 B | 391 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 16 kB | 16 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js tests/enable-deploy-e2e | Change |
| - | - | - | - |
| _buildManifest.js gzip | 460 B | 460 B | ✓ |
| Overall change | 460 B | 460 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js tests/enable-deploy-e2e | Change |
| - | - | - | - |
| index.html gzip | 531 B | 531 B | ✓ |
| link.html gzip | 545 B | 545 B | ✓ |
| withRouter.html gzip | 528 B | 528 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js tests/enable-deploy-e2e | Change |
| - | - | - | - |
| buildDuration | 22.5s | 22.7s | ⚠️ +262ms |
| buildDurationCached | 7.8s | 7.9s | ⚠️ +96ms |
| nodeModulesSize | 478 MB | 478 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js tests/enable-deploy-e2e | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 5.257 | 5.256 | 0 |
| / avg req/sec | 475.58 | 475.61 | +0.03 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.083 | 2.164 | ⚠️ +0.08 |
| /error-in-render avg req/sec | 1200.3 | 1155.36 | ⚠️ -44.94 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js tests/enable-deploy-e2e | Change |
| - | - | - | - |
| 925.HASH.js gzip | 178 B | 178 B | ✓ |
| framework-HASH.js gzip | 42.7 kB | 42.7 kB | ✓ |
| main-HASH.js gzip | 29.2 kB | 29.2 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 73.5 kB | 73.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js tests/enable-deploy-e2e | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js tests/enable-deploy-e2e | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 179 B | 179 B | ✓ |
| amp-HASH.js gzip | 311 B | 311 B | ✓ |
| css-HASH.js gzip | 324 B | 324 B | ✓ |
| dynamic-HASH.js gzip | 2.89 kB | 2.89 kB | ✓ |
| head-HASH.js gzip | 357 B | 357 B | ✓ |
| hooks-HASH.js gzip | 920 B | 920 B | ✓ |
| image-HASH.js gzip | 5.81 kB | 5.81 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 2.78 kB | 2.78 kB | ✓ |
| routerDirect..HASH.js gzip | 322 B | 322 B | ✓ |
| script-HASH.js gzip | 392 B | 392 B | ✓ |
| withRouter-HASH.js gzip | 317 B | 317 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 16.3 kB | 16.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js tests/enable-deploy-e2e | Change |
| - | - | - | - |
| _buildManifest.js gzip | 457 B | 457 B | ✓ |
| Overall change | 457 B | 457 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js tests/enable-deploy-e2e | Change |
| - | - | - | - |
| index.html gzip | 531 B | 531 B | ✓ |
| link.html gzip | 546 B | 546 B | ✓ |
| withRouter.html gzip | 528 B | 528 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
Commit: f3ed673dd316eec03235104a90e1b1f05f2213a8

File: .github/workflows/build_test_deploy.yml [hunk 1]
```diff
@@ -908,6 +908,46 @@ jobs:
...
+          node-version: 16
+          check-latest: true
+
+      - uses: actions/cache@v3
+        id: restore-build
+        with:
+          path: ./*
+          key: ${{ github.sha }}-${{ github.run_number }}
+
+      - run: npm i -g playwright-chromium@1.14.1 && npx playwright install-deps
```
Comment (styfle) [inline .github/workflows/build_test_deploy.yml:933]: Should this step have a name too?

File: .github/workflows/build_test_deploy.yml [hunk 2]
```diff
@@ -908,6 +908,46 @@ jobs:
...
+
+      - run: RESET_VC_PROJECT=true node scripts/reset-vercel-project.mjs
+        name: Reset test project
+
+      - run: NEXT_TEST_MODE=deploy node run-tests.js --type e2e
+        name: Run test/e2e (deploy)
+
+      - name: Upload test trace
+        if: always()
```
Comment (styfle) [inline .github/workflows/build_test_deploy.yml:942]: Is the if statement here necessary?

**Review by styfle — DISMISSED** [review]

File: .github/workflows/build_test_deploy.yml (hunk 1 above)
Comment (ijjk) [reply .github/workflows/build_test_deploy.yml:933]: Added in https://github.com/vercel/next.js/pull/37019/commits/dce8cb55d2c05bd8b3fade57afc3811c278e37c2

File: .github/workflows/build_test_deploy.yml (hunk 2 above)
Comment (ijjk) [reply .github/workflows/build_test_deploy.yml:943]: I believe this is still needed to force running a job after the previous job failed.

Comment (LOGOLFGODORD) [reply .github/workflows/build_test_deploy.yml:943]: Yes sir !!

**Pull request merged**